react-app-rss/src/pages/Form/form.tsx

```diff
@@ -0,0 +1,193 @@
+import React, { PureComponent, ReactNode } from 'react';
+import Card from '../../modules/components/card/card';
+import cls from './form.module.scss';
+
+interface FormData {
+  newImageURL: string | undefined | null;
+  weaponsArr:
+    | [
+        {
+          img: string;
+          name: string;
+          price: number;
+          caliber: string;
+          ammo: string;
+          damage: number;
+          distance: number;
+          reload: number;
+        }
+      ]
+    | [];
+}
+
+interface Elements {
+  name: React.RefObject<HTMLInputElement>;
+  price: React.RefObject<HTMLInputElement>;
+  caliber: React.RefObject<HTMLSelectElement>;
+  ammo: React.RefObject<HTMLInputElement>;
+  damage: React.RefObject<HTMLInputElement>;
+  distance: React.RefObject<HTMLInputElement>;
+  reload: React.RefObject<HTMLInputElement>;
+  img: React.RefObject<HTMLInputElement>;
+  date: React.RefObject<HTMLInputElement>;
+  radioYes: React.RefObject<HTMLInputElement>;
+  radioNo: React.RefObject<HTMLInputElement>;
+  check: React.RefObject<HTMLInputElement>;
+}
+class AddedForm extends PureComponent<object, FormData> {
+  formRef: React.RefObject<HTMLFormElement>;
+
+  elements: Elements;
+
+  constructor(props: object) {
+    super(props);
+    this.formRef = React.createRef();
+
+    this.state = {
+      newImageURL: '',
+      weaponsArr: [],
+    };
+
+    this.elements = {
+      name: React.createRef(),
+      price: React.createRef(),
+      caliber: React.createRef(),
+      ammo: React.createRef(),
+      damage: React.createRef(),
+      distance: React.createRef(),
+      reload: React.createRef(),
+      img: React.createRef(),
+      date: React.createRef(),
+      radioYes: React.createRef(),
+      radioNo: React.createRef(),
+      check: React.createRef(),
+    };
+
+    this.handleUploadImage = this.handleUploadImage.bind(this);
+  }
+
+  handleUploadImage() {
+    const uploadImage = this.elements.img.current?.files![0];
+    const imageUrl = URL.createObjectURL(uploadImage as Blob | MediaSource);
+    this.setState({ newImageURL: imageUrl });
+  }
+
+  showResult() {
+    const { weaponsArr } = this.state;
+    const { newImageURL } = this.state;
+    (weaponsArr as [object]).push({
+      img: newImageURL,
+      name: this.elements.name.current?.value,
+      price: Number(this.elements.price.current?.value),
+      caliber: this.elements.caliber.current?.value,
+      ammo: this.elements.ammo.current?.value,
+      damage: Number(this.elements.damage.current?.value),
+      distance: Number(this.elements.distance.current?.value),
+      reload: Number(this.elements.distance.current?.value),
+    });
+    this.setState({ weaponsArr });
```

Comment:
potential race condition here, consider using showResult() {
  this.setState(prevState => {.... instead.